src/box/lua/connpool.lua

```diff
@@ -75,21 +75,34 @@ local function connect_to_candidates(candidates)
     local delay = WATCHER_DELAY
     local conn_opts = {connect_timeout = WATCHER_TIMEOUT}
     local connected_candidates = {}
+
+    local connection_fibers = {}
+
     for _, instance_name in pairs(candidates) do
-        local time_connect_end = clock.monotonic() + WATCHER_TIMEOUT
-        local ok, conn = pcall(connect, instance_name, conn_opts)
-        -- If state is not 'active', conn:mode() cannot become not nil.
-        if ok and conn.state == 'active' then
-            while conn:mode() == nil and clock.monotonic() < time_connect_end do
-                fiber.sleep(delay)
+        local connection_fiber = fiber.new(function()
```

Comment:
I wouldn't create a new fiber if the connection is alredy in a good state.

Reply:
That's a good idea. Done.

Reply:
UPD: Now, not creating a new fibers at all.

Reply:
Irrelevant now, closing.